query-sql/src/lib.rs

```diff
@@ -290,19 +322,15 @@ impl QueryFragment for SelectQuery {
     fn push_sql(&self, out: &mut QueryBuilder) -> BuildQueryResult {
         out.push_sql("SELECT ");
         self.projection.push_sql(out)?;
-
-        out.push_sql(" FROM ");
         self.from.push_sql(out)?;
 
-        if self.constraints.is_empty() {
-            return Ok(());
+        if !self.constraints.is_empty() {
```

Comment:
Is it ever the case that `self.constraints` will be non-empty but `from` will be `None`?  If it is, you might consider trying to simplify the SQL produced from `self.projection` and `self.constraints`.  This is very much up to you, though, since I doubt you want to make the query itself represent that it is from `None`.